File: api/UserManagement/UserManagement/Models/DTO/UserDto.cs
namespace UserManagement.Models.DTO
{
    public class UserDto
    {
        public Guid Id { get; set; }

        public required string FirstName { get; set; }

        public required string LastName { get; set; }

        public required string Email { get; set; }

        public string? Phone { get; set; }

        public required RoleDto Role { get; set; }

        public required string Username { get; set; }

        public required List<UserPermissionDto> Permissions { get; set; }
    }
}

File: app/user-management/src/app/models/user-dto.ts
export interface UserDto {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  role: RoleDto;
  username: string;
  permissions: UserPermissionDto[];
}
export interface RoleDto {
  roleId: string;
  roleName: string;
}
export interface UserPermissionDto {
  permissionId: string;
  permissionName: string;
}
